ddtrace/contrib/kombu/patch.py

```diff
@@ -0,0 +1,91 @@
+# 3p
+import kombu
+import wrapt
+
+# project
+from ...pin import Pin
+from ...ext import kombu as kombux
+from ...utils.wrappers import unwrap
+from ...propagation.http import HTTPPropagator
+from .util import (
+    get_exchange_from_args,
+    get_body_length_from_args,
+    get_routing_key_from_args,
+    _extract_conn_tags
+)
+
+HEADER_POS = 4
+propagator = HTTPPropagator()
+
+
+def patch():
+    """Patch the instrumented methods
+
+    This duplicated doesn't look nice. The nicer alternative is to use an ObjectProxy on top
+    of Kombu. However, it means that any "import kombu.Connection" won't be instrumented.
+    """
+    if getattr(kombu, '_datadog_patch', False):
+        return
+    setattr(kombu, '_datadog_patch', True)
+
+    _w = wrapt.wrap_function_wrapper
+    _w('kombu', 'Producer._publish', traced_publish)
+    _w('kombu', 'Consumer.receive', traced_receive)
+    Pin(service="kombu", app="kombu", app_type="custom").onto(kombu.messaging.Producer)
+    Pin(service="kombu", app="kombu", app_type="custom").onto(kombu.messaging.Consumer)
+
+
+def unpatch():
+    if getattr(kombu, '_datadog_patch', False):
+        setattr(kombu, '_datadog_patch', False)
+        unwrap(kombu.Producer, '_publish')
+        unwrap(kombu.Consumer, 'receive')
+
+#
+# tracing functions
+#
+
+
+def traced_receive(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    # Signature only takes 2 args: (body, message)
+    message = args[1]
+    context = propagator.extract(message.headers)
+    # only need to active the new context if something was propagated
+    if context.trace_id:
+        pin.tracer.context_provider.activate(context)
+    with pin.tracer.trace('kombu.Consumer.receive', service=pin.service, span_type='kombu') as s:
+        # run the command
+        exchange = message.delivery_info['exchange']
+        s.resource = exchange
+        s.set_tag(kombux.EXCHANGE, exchange)
+
+        s.set_tags(_get_tags(message.channel.connection))
+        s.set_tag(kombux.ROUTING_KEY, message.delivery_info['routing_key'])
+        return func(*args, **kwargs)
+
+
+def traced_publish(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    with pin.tracer.trace('kombu.Producer._publish', service=pin.service, span_type='kombu') as s:
```

Comment:
The name of the operation couldn't be simply `kombu.publish` instead of `kombu.Producer._publish`? I think it's more readable from users stand point. What do you think?